editor/src/core/shared/element-path-tree.ts

```diff
@@ -54,6 +59,56 @@ export function buildTree(elementPaths: Array<ElementPath>): ElementPathTreeRoot
   return result
 }
 
+export function reorderTree(
+  tree: ElementPathTree,
+  metadata: ElementInstanceMetadataMap,
+): ElementPathTree {
+  const element = MetadataUtils.findElementByElementPath(metadata, tree.path)
+  if (element == null) {
+    return tree
+  } else {
+    return foldEither(
+      () => {
+        return tree
+      },
+      (elementChild) => {
+        switch (elementChild.type) {
+          case 'JSX_ELEMENT': {
+            const allChildrenAreElements = elementChild.children.every(isJSXElement)
+            if (allChildrenAreElements) {
```

Comment:
It seems we don't need to restrict this to cases where all children are JSX Elements? Couldn't we just skip over children that aren't?

Reply:
If you have an element with children of element, arbitrary block, element then it can end up doing odd things with reordering the last element into the middle of the generated elements produced by the arbitrary block.